configs/containers/docker-ubuntu-oneapi-impi.yaml

```diff
@@ -0,0 +1,320 @@
+spack:
+  concretizer:
+    unify: true
+
+  config:
+    checksum: false
+    build_jobs: 2
+    connect_timeout: 60
+
+  compilers:
+  - compiler:
+      spec: gcc@=13.4.0
+      paths:
+        cc: /usr/bin/gcc
+        cxx: /usr/bin/g++
+        f77: /usr/bin/gfortran
+        fc: /usr/bin/gfortran
+      flags: {}
+      operating_system: ubuntu24.04
+      target: x86_64
+      modules: []
+      environment: {}
+      extra_rpaths: []
+  - compiler:
+      spec: oneapi@=2024.2.1
+      paths:
+        cc: /opt/intel/oneapi/compiler/2024.2/bin/icx
+        cxx: /opt/intel/oneapi/compiler/2024.2/bin/icpx
+        f77: /opt/intel/oneapi/compiler/2024.2/bin/ifx
+        fc: /opt/intel/oneapi/compiler/2024.2/bin/ifx
+        #f77: /opt/intel/oneapi/compiler/2024.2/bin/ifort
+        #fc: /opt/intel/oneapi/compiler/2024.2/bin/ifort
```

Comment:
I don't know which is the preferred compiler here - @climbfuji 's August 15 comment in #912 and the checked in code in `configs/sites/tier2/blackpearl` implies that ifort might work better, but I did build with `ifx` so maybe some of that info is out of date.

Reply:
Thanks. That sounds like a good balance. I switched this back to ifort and it still compiles with no other changes.